### ISSUE TITLE
sklearn set_output

### ISSUE THREAD

**Issue by mglisse** [GUDHI/gudhi-devel#1015]
Since version 1.2, scikit-learn has a set_output API that can make transformers output pandas dataframes.

- [ ] Investigate how that affects us.

**Comment by mglisse** [comment]
@martinroyer suggests adding to `RipsPersistence`
```python
def get_feature_names_out(self):
    return [f"H{i}" for i in self.homology_dimensions]
```
so that `RipsPersistence(homology_dimensions=[0, 2]).set_output(transform="pandas").fit_transform([X])` can output a pandas dataframe with column labels "H0" and "H2".
